src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3374,7 +3378,9 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
                   // the overloaded type into a single function type from which `typedFunction`
                   // can derive the argument type for `x` in the function literal above
                   val argPt =
-                    if (argPtAlts.nonEmpty && treeInfo.isFunctionMissingParamType(tree)) functionProto(argPtAlts)
+                    if (argPtAlts.isEmpty) WildcardType
+                    else if (treeInfo.isFunctionMissingParamType(tree)) functionProto(argPtAlts)
+                    else if (treeInfo.isPartialFunctionMissingParamType(tree)) partialFunctionProto(argPtAlts)
```

Comment:
A difference is that now the compiler will generate a PartialFunction even if the selected alternative has a Function parameter type
```
object T {
  def m(f: Object => Int, s: String): Int = 1
  def m(f: String => Int, s: Object): Int = 1
}

T.m({ case x => 2 }, null)
```
generates
```
T.m(({
          @SerialVersionUID(value = 0) final <synthetic> class $anonfun extends scala.runtime.AbstractPartialFunction[Object,Int] with Serializable {
            def <init>(): <$anon: Object => Int> = {
              $anonfun.super.<init>();
              ()
            };
            final override def applyOrElse[A1 <: Object, B1 >: Int](x1: A1, default: A1 => B1): B1 = ((x1.asInstanceOf[Object]: Object): Object @unchecked) match {
              case (x @ _) => 2
              case (defaultCase$ @ _) => default.apply(x1)
            };
            final def isDefinedAt(x1: Object): Boolean = ((x1.asInstanceOf[Object]: Object): Object @unchecked) match {
              case (x @ _) => true
              case (defaultCase$ @ _) => false
            }
          };
          new $anonfun()
        }: PartialFunction[Object,Int]), null);
```

In 2.13.0-M1

```
T.m(((x0$1: Object) => x0$1 match {
          case (x @ _) => 2
        }), null);
```